java-manta-client-unshaded/src/test/java/com/joyent/manta/client/crypto/MultiCryptoProviderDecryptionTest.java

```diff
@@ -0,0 +1,309 @@
+                MantaClient.SEPARATOR + cipherName + ".headers");
+        final Map<String, Object> headerMap = parseHeaderFileInfoStoreInMap(headerURL);
+
+        /* Populate header values */
+
+        if (!cipherDetails.isAEADCipher()) {
+            getResponseHttpHeaders.put(ENCRYPTION_HMAC_TYPE,
+                    headerMap.get(ENCRYPTION_HMAC_TYPE));
+
+            getResponseHttpHeaders.put(ENCRYPTION_METADATA_HMAC,
+                    headerMap.get(ENCRYPTION_HMAC_TYPE));
+        }
+        else {
+            getResponseHttpHeaders.put(ENCRYPTION_METADATA_AEAD_TAG_LENGTH,
+                    headerMap.get(ENCRYPTION_METADATA_AEAD_TAG_LENGTH));
+        }
+
+
+        final long contentLength = Long.parseLong(headerMap.get(CONTENT_LENGTH).toString());
+        getResponseHttpHeaders.setContentLength(contentLength);
+
+        getResponseHttpHeaders.put(ENCRYPTION_IV,
+                headerMap.get(ENCRYPTION_IV));
+
+        getResponseHttpHeaders.put(ENCRYPTION_METADATA_IV,
+                headerMap.get(ENCRYPTION_METADATA_IV));
+
+        getResponseHttpHeaders.put(ENCRYPTION_KEY_ID,
+                headerMap.get(ENCRYPTION_KEY_ID));
+
+        getResponseHttpHeaders.put(ENCRYPTION_CIPHER,
+                headerMap.get(ENCRYPTION_CIPHER));
+
+        getResponseHttpHeaders.put(ENCRYPTION_PLAINTEXT_CONTENT_LENGTH,
+                headerMap.get(ENCRYPTION_PLAINTEXT_CONTENT_LENGTH));
+
+        getResponseHttpHeaders.put(ENCRYPTION_AEAD_TAG_LENGTH,
+                headerMap.get(ENCRYPTION_AEAD_TAG_LENGTH));
+
+
+        getResponseHttpHeaders.put(COMPUTED_MD5,
+                headerMap.get(ENCRYPTION_METADATA_AEAD_TAG_LENGTH));
+
+        getResponseHttpHeaders.put(ENCRYPTION_METADATA_AEAD_TAG_LENGTH,
+                headerMap.get(ENCRYPTION_METADATA_AEAD_TAG_LENGTH));
+
+        getResponseHttpHeaders.put(ENCRYPTION_TYPE,
+                headerMap.get(ENCRYPTION_TYPE));
+
+        getResponseHttpHeaders.setContentMD5(headerMap.get(CONTENT_MD5).toString());
+        getResponseHttpHeaders.setContentType(headerMap.get(CONTENT_TYPE).toString());
+
+    }
+
+    private static SecretKey generateSecretKey(final SupportedCipherDetails cipherDetails) {
+        int keySizeInBytes = cipherDetails.getKeyLengthBits() >> 3; // convert bits to bytes
+        byte[] secret = Arrays.copyOfRange(SECRET_KEY, 0, keySizeInBytes);
+        return SecretKeyUtils.loadKey(secret, cipherDetails);
+    }
+
+    private static void assertByteArrayEquals(final byte[] actual, final byte[] expected) {
+        if (!Arrays.equals(actual, expected)) {
+            String msg = String.format("Array values are not equal\n"
+                            + "Expected length: %d Actual length: %d\n"
+                            + "Expected: %s\n"
+                            + "Actual:   %s",
+                    expected.length, actual.length,
+                    Hex.toHexString(expected),
```

Comment:
Prefer the `org.apache.commons.codec.binary.Hex.encodeHexString` utility method over the bouncy castle method. We should search for the code that uses the Bouncy Castle method and switch it over to make everything uniform.